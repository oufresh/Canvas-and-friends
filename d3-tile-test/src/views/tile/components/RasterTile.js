import React from 'react';
import { TILE_SIZE } from '../../../modules/maps/utils/constants';

type TilePropsType = {
    key: string,
    url: string,
    x: number,
    y: number,
    width?: number,
    height?: number
};

export const defaultHeight: number = TILE_SIZE;
export const defaultWidth: number = TILE_SIZE;

const RasterTile = (props: TilePropsType) => {
    //console.log("RasterTile url: " + props.url);
    const h = props.height ? props.height : defaultHeight;
    const w = props.width ? props.width : defaultWidth;

    return <image href={props.url} width={w}
        height={h} x={props.x*w} y={props.y*h} />;
};

export default RasterTile;