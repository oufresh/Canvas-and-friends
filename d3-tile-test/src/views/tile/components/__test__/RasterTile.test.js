//@flow
import React from 'react';
import { shallow } from 'enzyme';
import RasterTile from '../RasterTile';

describe('RasterTile Component', () => {

    it('renders without a crash with preloaded svg content', () => {
        const component = shallow(<RasterTile key="test" x={0} y={0} z={0} url={"testurl"}/>);
        expect(component).toMatchSnapshot();
    });

});