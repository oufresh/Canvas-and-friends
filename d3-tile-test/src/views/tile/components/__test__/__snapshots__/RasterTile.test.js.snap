// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RasterTile Component renders without a crash with preloaded svg content 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RasterTile
    url="testurl"
    x={0}
    y={0}
    z={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "height": 256,
      "href": "testurl",
      "width": 256,
      "x": 0,
      "y": 0,
    },
    "ref": null,
    "rendered": null,
    "type": "image",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "height": 256,
        "href": "testurl",
        "width": 256,
        "x": 0,
        "y": 0,
      },
      "ref": null,
      "rendered": null,
      "type": "image",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
